DisCatSharp.ApplicationCommands/Workers/ApplicationCommandWorker.cs

```diff
@@ -0,0 +1,377 @@
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using DisCatSharp.Entities;
+using DisCatSharp.Enums;
+
+namespace DisCatSharp.ApplicationCommands
+{
+    /// <summary>
+    /// Represents a <see cref="CommandWorker"/>.
+    /// </summary>
+    internal class CommandWorker
+    {
+        /// <summary>
+        /// Parses context menu application commands.
+        /// </summary>
+        /// <param name="type">The type.</param>
+        /// <param name="methods">List of method infos.</param>
+        /// <param name="guildid">The optional guild id.</param>
+        /// <param name="translator">The optional command translations.</param>
+        /// <returns>Too much.</returns>
+        internal static Task<Tuple<List<DiscordApplicationCommand>, List<KeyValuePair<Type, Type>>, List<ContextMenuCommand>>> ParseContextMenuCommands(Type type, IEnumerable<MethodInfo> methods, ulong? guildid = null, List<CommandTranslator> translator = null)
+        {
+            List<DiscordApplicationCommand> Commands = new();
+            List<KeyValuePair<Type, Type>> CommandTypeSources = new();
+            List<ContextMenuCommand> contextMenuCommands = new();
+
+
+            foreach (var contextMethod in methods)
+            {
+                var contextAttribute = contextMethod.GetCustomAttribute<ContextMenuAttribute>();
+
+                DiscordApplicationCommandLocalization NameLocalizations = null;
+
+                var command_translation = translator?.Single(c => c.Name == contextAttribute.Name && c.Type == contextAttribute.Type);
+                if (command_translation != null)
+                    NameLocalizations = command_translation.NameTranslations;
+
+                var command = new DiscordApplicationCommand(contextAttribute.Name, null, null, contextAttribute.DefaultPermission, contextAttribute.Type, NameLocalizations);
+
+                var parameters = contextMethod.GetParameters();
+                if (parameters.Length == 0 || parameters == null || !ReferenceEquals(parameters.FirstOrDefault()?.ParameterType, typeof(ContextMenuContext)))
+                    throw new ArgumentException($"The first argument must be a ContextMenuContext!");
+                if (parameters.Length > 1)
+                    throw new ArgumentException($"A context menu cannot have parameters!");
+
+                contextMenuCommands.Add(new ContextMenuCommand { Method = contextMethod, Name = contextAttribute.Name });
+
+                Commands.Add(command);
+                CommandTypeSources.Add(new KeyValuePair<Type, Type>(type, type));
+            }
+
+            return Task.FromResult(Tuple.Create(Commands, CommandTypeSources, contextMenuCommands));
+        }
```

Comment:
I know you have taken a liking to tuples but having the named version of this is much better than referencing this return type as Item1, Item2, Item3, etc.